### PR TITLE
feat(tracer): Add function params to trace

### PR DESCRIPTION
**Pull request by smanilov**
The `variables` in `StackFrame` are now sorted before construction of the frame. This is an unenforced invariant of the type. The sorting is thus not needed anymore after the field is accessed (previously, done in `fn register_variables`).

The new field of `StackFrame` -- `function_param_indexes` -- is a vector of indexes in this sorted vector (`variables`). It lists which variables are function parameters.

`stack_trace` is renamed to `stack_frames` for increased symmetry of the code (it's of type `Vec<StackFrame>`)

Finally, instead of an empty vec[], actual args are passed to `register_call`, which are built on top of the other changes mentioned.

One question that remains open is whether `register_value` needs to be called in `convert_params_to_args_vec` or not. This is noted in a comment and will be addressed at a later point.

### CODE REVIEW

File: tooling/tracer/src/tracer_glue.rs [hunk 1]
```diff
@@ -85,7 +83,21 @@ pub(crate) fn register_call(tracer: &mut Tracer, location: &SourceLocation, fram
...
+
+/// Extracts the relevant information from the given `frame` to construct a vector of `ArgRecord`
+/// that the `Tracer` interface expects when registering function calls.
+fn convert_params_to_args_vec(tracer: &mut Tracer, frame: &StackFrame) -> Vec<ArgRecord> {
+    let mut result = Vec::new();
+    for param_index in &frame.function_param_indexes {
+        let variable = &frame.variables[*param_index];
+        // TODO(stanm): maybe don't duplicate values?
+        let value_record = register_value(tracer, &variable.value, &variable.typ);
+        result.push(ArgRecord { name: String::from(&variable.name), value: value_record });
```
Comment (alehander92) [inline tooling/tracer/src/tracer_glue.rs:98]: forgive me, I could warn you, but I delayed that: we updated to `0.5.9`, and we don't support `ArgRecord` anymore: we use similar structures and  mechanisms that we use for the other local variables now: so one should just do `tracer.arg(&variable.name, value_record)` and add that to the `result`

Comment (smanilov) [reply tooling/tracer/src/tracer_glue.rs:98]: Thanks for the note. I will update to 0.5.9 and will change this code before merging.

**Review by Aristotelis2002 — APPROVED** [review]
LGTM.

File: tooling/tracer/src/tracer_glue.rs (hunk 1 above)
Comment (smanilov) [reply tooling/tracer/src/tracer_glue.rs]: Done.

**Pull request merged**